tasks/add_picklist_values.py

```diff
@@ -0,0 +1,339 @@
+        for value in values:
+            # ignore any existing picklist values
+            if value in values_added:
+                self.logger.info(f'{value} is already a picklist value on {field}. Skipping over...')
+
+            else:
+                picklist_values_xml += picklist_value_template.format(
+                    name = value, 
+                    default = False, 
+                    label = value
+                )
+
+        # add "Other", if it exists
+        picklist_values_xml += other_picklist_xml
+
+        # add the picklist values to the record types, if any were specified
+        if picklist_record_types:
+            for rt in picklist_record_types:
+                record_type_picklist_values_xml = ""
+                values_added_to_record_type = []
+
+                # add back the existing picklist values assigned to the record type
+                for picklist in rt["Metadata"]["picklistValues"]:
+                    if picklist["picklist"] == field:
+                        for picklist_value in picklist["values"]:
+                            values_added_to_record_type.append(picklist_value["valueName"])
+                            record_type_picklist_values_xml += record_type_picklist_value_template.format(
+                                name = picklist_value["valueName"],
+                                default = picklist_value["default"]
+                            )
+                        break
+
+                # assign the new picklist values to the record type
+                for value in values:
+                    # ignore any existing picklist values
+                    if value in values_added_to_record_type:
+                        continue
+
+                    record_type_picklist_values_xml += record_type_picklist_value_template.format(
+                        name = value,
+                        default = False
+                    )
+
+                record_type_picklist_xml += record_type_picklist_template.format(
+                    name = rt["FullName"].split('.')[1], # removes the SObject from FullName
+                    description = rt["Metadata"]["description"],
+                    label = rt["Name"], 
+                    field = field,
+                    picklist_values = record_type_picklist_values_xml
+                )
+
+        sorted_picklist = validated_field["valueSet"]["valueSetDefinition"]["sorted"]
+        if "sorted" in self.options and self.options["sorted"] == "True":
+            sorted_picklist = True
+
+        # combine it all together
+        object_xml = object_template.format(
+            field = field,
+            description = validated_field["description"],
+            helpText = validated_field["inlineHelpText"],
+            label = validated_field["label"],
+            sorted = sorted_picklist,
+            picklist_values = picklist_values_xml,
+            record_types = record_type_picklist_xml
+        )
```

Comment:
Building XML through simple string substitution like this makes me nervous. What if the label, description, or help text contain characters like `<` and `>` that will be interpreted as XML?

Reply:
I'll test this and get back to you on this. I'm curious how Salesforce returns such characters in the Tooling API call (validated_field represents the Metadata of the CustomField retrieved via the Tooling API)

Reply:
Fixed this by using the escape() function

Reply:
That's good enough for now. In the long run I'd like CumulusCI to have a utility to help with building metadata XML so that you don't have to remember to call escape each time.